arm-datalake-store/filesystem/2016-11-01/swagger/filesystem.json

```diff
@@ -1419,30 +1521,128 @@
         "FileStatus": {
           "$ref": "#/definitions/FileStatusProperties",
           "readOnly": true,
-          "description": "Gets the file status object associated with the specified path."
+          "description": "the file status object associated with the specified path."
         }
       },
       "description": "Data Lake Store filesystem file status information response."
     },
-    "FileInfoResult": {
+    "AdlsIllegalArgumentException": {
+      "x-ms-discriminator-value": "IllegalArgumentException",
+      "allOf": [ 
+        { 
+          "$ref":  "#/definitions/AdlsRemoteException" 
+        } 
+      ],
+      "description": "A WebHDFS exception thrown indicating that one more arguments is incorrect. Thrown when a 400 error response code is returned (bad request)."
```

Comment:
According to the WebHDFS spec "IllegalArgumentExceptions" are only thrown with a corresponding 400 error. If our service does not honor the webhdfs spec here, its a bug (same with the other comments about specific return codes).